### PR TITLE
Fast-forward continuous-benchmarking branch

### PR DESCRIPTION
**Pull request by kk-min**
This PR fast forwards the `continuous-benchmarking` branch to encapsulate developments in `feature-serverless-framework-deployment`.

## Changes
- All merged code in `feature-serverless-framework-deployment`
- Add warm flag `w` for warm invocation experiments that drop cold start first result
- Change lambda endpoint for uploading benchmarking results

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,10 +81,76 @@ Finally, we look at the procedural steps adopted by the framework:
...
+| Go Runtime                                   | Yes        | No                                       | No               | Yes     | Yes                                      | ??**                                     |
+| Java Runtime                                 | Yes        | No                                       | No               | ??**    | No                                       | ??**                                     |
+| Node JS                                      | Yes        | Yes                                      | No               | Yes     | Yes                                      | Yes                                      |
+| HTTP trigger                                 | Yes        | Yes                                      | No               | Yes     | Yes                                      | Yes                                      |
+| Resource Management (e.g. buckets databases) | Yes        | Yes (at least it looks like it)          | No               | ??**    | ??**                                     | ??**                                     |
```
Comment (dilinade) [inline README.md:147]: We might need to change the readme a bit, but I think this is okay for now.
@kk-min Can you create a separate issue for this? So that we can attend to it later?

File: continuous-benchmarking/experiments/cold-function-invocations/cold-baseline-cloudflare.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	  "Handler": "dist/main.js",
+      "Bursts": 50,
+      "BurstSizes": [
+        1
+      ],
+      "IATSeconds": 600,
+      "DesiredServiceTimes": [
+        "0ms"
+      ],
+      "FunctionImageSizeMB": 24,
```
Comment (dilinade) [inline continuous-benchmarking/experiments/cold-function-invocations/cold-baseline-cloudflare.json:18]: Do we need this parameter here? Function Image Size?

File: src/setup/test/test_snapstart.yml
Comment (dilinade) [inline src/setup/test/test_snapstart.yml:1]: Do we need the snapstart files?

File: src/setup/test/serverless-config_test.go [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+	// Generate YAML content from the expected Serverless struct
+	expectedData, err := os.ReadFile("test.yml")
+	assert.NoError(err, "Error marshaling expected data")
+
+	// Compare the contents byte by byte
+	assert.True(bytes.Equal(expectedData, actualData), "YAML content mismatch")
+
+}
+
+func TestCreateServerlessConfigFileSnapStart(t *testing.T) {
```
Comment (dilinade) [inline src/setup/test/serverless-config_test.go:180]: Snapstart?

File: src/setup/deployment/raw-code/serverless/cloudflare/hellopy/__target__/org.transcrypt.__runtime__.map
Comment (dilinade) [inline src/setup/deployment/raw-code/serverless/cloudflare/hellopy/__target__/org.transcrypt.__runtime__.map:1]: Just out of curiosity, are these files required and not dynamically generated?

**Review by dilinade — APPROVED** [review]

File: src/setup/test/test_snapstart.yml
Comment (kk-min) [reply src/setup/test/test_snapstart.yml:1]: Yes, this file is needed for asserting that the correct yml is generated in the test `TestCreateServerlessConfigFileSnapStart`.

File: src/setup/test/serverless-config_test.go (hunk 3 above)
Comment (kk-min) [reply src/setup/test/serverless-config_test.go:180]: The official term by AWS seems to use [SnapStart with 2 capital S'es.](https://docs.aws.amazon.com/lambda/latest/dg/snapstart.html).

File: src/setup/deployment/raw-code/serverless/cloudflare/hellopy/__target__/org.transcrypt.__runtime__.map
Comment (kk-min) [reply src/setup/deployment/raw-code/serverless/cloudflare/hellopy/__target__/org.transcrypt.__runtime__.map:1]: For deployment the actual file needed is in `hellopy/dist/main.js`, I'll clean this up. For now, to make things simpler, we assume that the user will transpile the .py file to .main.js, since [Cloudflare's official example](https://github.com/cloudflare/python-worker-hello-world) suggests it, so users can follow along. I'll highlight this when writing the wiki for benchmarking with Cloudflare.

Another potential issue is that transcrypt does not necessarily support all python libraries, so users need to exercise caution when transpiling and ensure their code only contains the supported libraries by transcrypt. If we transpile it for them, they may not know what went wrong if unsupported libraries are used.

File: README.md (hunk 1 above)
Comment (kk-min) [reply README.md:147]: I've created an issue here: #312
